lms/djangoapps/course_blocks/transformers/utils.py

```diff
@@ -18,6 +18,43 @@ def get_field_on_block(block, field_name, default_value=None):
     return default_value
 
 
+def collect_unioned_set_field(block_structure, transformer, merged_field_name, base_case_predicate):
+    """
+    Recursively union a set field on the block structure.
+
+    If a block matches base_case_predicate, it will be added to the
+    result set. This (potentially empty) set is unioned with the sets
+    contained in merged_field_name for all parents of the block.
+
+    This set union operation takes place during a topological traversal
+    of the block_structure, so all sets are inherited by descendants.
+
+    Parameters:
+        block_structure: BlockStructure to traverse
+        transformer: transformer that will be used for get_ and
+            set_transformer_block_field
+        merged_field_name: name of the field to store
+        base_case_predicate: a unary lambda that returns true if a given
```

Comment:
I'm wondering if the name "base_case_predicate" is valid here.. since it isn't really a "base case" anymore - no longer a recursive function.